### PR TITLE
Add form view for creating and editing documents

### LINKED ISSUE #790: Make JSON-form-view great again?
The JSON form view might actually not be that useless in the end...

### PR DESCRIPTION
**Pull request by Njuelle**
## What does this PR do ?
fix #790 
Adds the ability to create or edit a document with a new form view.

The automatic generation of the form is made from a schema generated from the mapping ([more info](https://vue-generators.gitbook.io/vue-generators/usage)).

Currently `arrays` are not supported, however a warning is displayed to the user, inviting him to use the JSON view instead.

Also, the JSON view and the Form view are synchronized with each other.

image

### CODE REVIEW

**Review by jenow — COMMENTED** [review]
When I have a nested object, it shows me a textare with just `{}` inside. It would be good to at least have the mapping. 
a

**Review by alexandrebouthinon — APPROVED** [review]

**Comment by berthieresteban** [comment]
Just a question
Since dates are timestamp, shouldn't we allow user to enter date and hour by adding a `<b-timepicker>` for example ? 
image

**Review by berthieresteban — APPROVED** [review]

File: src/services/formSchema.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  short: 'number',
+  byte: 'number',
+  double: 'number',
+  float: 'number',
+  half_float: 'number',
+  scaled_float: 'number',
+  keyword: 'text',
+  wildcard: 'text',
+  constant_keyword: 'text',
+  date: 'date'
```
Comment (berthieresteban) [inline src/services/formSchema.ts:56]:
```suggestion
  date: 'datetime-local'
```

**Review by berthieresteban — CHANGES_REQUESTED** [review]

**Review by xbill82 — COMMENTED** [review]
All good so far, let's wait for the date improvement next week. I'll change this to a WIP PR.

**Comment by Njuelle** [comment]
DateTime form input :
image

**Comment by Njuelle** [comment]
Json input form with mapping :
image

**Comment by berthieresteban** [comment]
Ci tests fails only on the date/time pickers of the form view, we are asserting timestamps and there is a time difference of 1h.
It might be because of the runners timezone.

**Review by alexandrebouthinon — CHANGES_REQUESTED** [review]

**Review by jenow — APPROVED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

**Pull request merged**